{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": [
        "nG9yEcd7CVHp"
      ],
      "authorship_tag": "ABX9TyMMRle3G+pBXwVwYPNon/Rp",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/akrysm/rnaseq/blob/main/RNA_seq.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#準備(GoogleDrive, インストール, ディレクトリ・フォルダ設定, コンテンツ確認)"
      ],
      "metadata": {
        "id": "IbpgrPyuCf15"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "EMSBU2lYgzcB",
        "outputId": "8cb7d169-8e51-4004-9442-59003f6f5165"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ],
      "source": [
        "#@title Google Driveをマウント\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title インストール\n",
        "import gzip\n",
        "import locale\n",
        "import multiqc\n",
        "import os\n",
        "import shutil\n",
        "import subprocess\n",
        "import time\n",
        "\n",
        "print(multiqc.__version__)\n",
        "\n",
        "\n",
        "!apt-get -qq install hisat2\n",
        "!apt-get -qq install samtools\n",
        "!apt-get -qq install subread\n",
        "!apt-get -qq install tree\n",
        "os.environ['PATH'] += \":/usr/local/bin\"\n",
        "!hisat2 --version\n",
        "!samtools --version\n",
        "!subread --version\n",
        "!tree --version"
      ],
      "metadata": {
        "id": "0dlHbUxwoadZ",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "outputId": "3e8bb73a-1015-4395-c8a5-5030cf82834e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1.22.dev0\n",
            "Selecting previously unselected package bcftools.\n",
            "(Reading database ... 121852 files and directories currently installed.)\n",
            "Preparing to unpack .../bcftools_1.13-1_amd64.deb ...\n",
            "Unpacking bcftools (1.13-1) ...\n",
            "Selecting previously unselected package hisat2.\n",
            "Preparing to unpack .../hisat2_2.2.1-3_amd64.deb ...\n",
            "Unpacking hisat2 (2.2.1-3) ...\n",
            "Selecting previously unselected package python3-hisat2.\n",
            "Preparing to unpack .../python3-hisat2_2.2.1-3_all.deb ...\n",
            "Unpacking python3-hisat2 (2.2.1-3) ...\n",
            "Setting up bcftools (1.13-1) ...\n",
            "Setting up hisat2 (2.2.1-3) ...\n",
            "Setting up python3-hisat2 (2.2.1-3) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Selecting previously unselected package subread.\n",
            "(Reading database ... 121953 files and directories currently installed.)\n",
            "Preparing to unpack .../subread_2.0.3+dfsg-1_amd64.deb ...\n",
            "Unpacking subread (2.0.3+dfsg-1) ...\n",
            "Setting up subread (2.0.3+dfsg-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Selecting previously unselected package tree.\n",
            "(Reading database ... 121988 files and directories currently installed.)\n",
            "Preparing to unpack .../tree_2.0.2-1_amd64.deb ...\n",
            "Unpacking tree (2.0.2-1) ...\n",
            "Setting up tree (2.0.2-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "/usr/bin/hisat2-align-s version 2.2.1\n",
            "64-bit\n",
            "Built on Debian\n",
            "28 September 2021\n",
            "Compiler: gcc version 11.2.0 (Ubuntu 11.2.0-7ubuntu2) \n",
            "Options: -O3   -funroll-loops -g3 -Wdate-time -D_FORTIFY_SOURCE=2 -std=c++11\n",
            "Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}\n",
            "samtools 1.13\n",
            "Using htslib 1.13+ds\n",
            "Copyright (C) 2021 Genome Research Ltd.\n",
            "\n",
            "Samtools compilation details:\n",
            "    Features:       build=configure curses=yes \n",
            "    CC:             gcc\n",
            "    CPPFLAGS:       -frelease  -Wdate-time -D_FORTIFY_SOURCE=2\n",
            "    CFLAGS:         -g -O2 -ffile-prefix-map=�BUILDPATH�=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security\n",
            "    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now\n",
            "    HTSDIR:         \n",
            "    LIBS:           \n",
            "    CURSES_LIB:     -lcurses\n",
            "\n",
            "HTSlib compilation details:\n",
            "    Features:       build=configure plugins=yes, plugin-path=/usr/local/lib/htslib:/usr/local/libexec/htslib:: libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes htscodecs=1.1.1\n",
            "    CC:             gcc\n",
            "    CPPFLAGS:       -I. -DSAMTOOLS=1 -Wdate-time -D_FORTIFY_SOURCE=2\n",
            "    CFLAGS:         -g -O2 -ffile-prefix-map=/build/htslib-TQtOKr/htslib-1.13+ds=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -flto -fvisibility=hidden -flto -fvisibility=hidden\n",
            "    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,-flto -fvisibility=hidden -Wl,-flto -fvisibility=hidden\n",
            "\n",
            "HTSlib URL scheme handlers present:\n",
            "    built-in:\t preload, data, file\n",
            "    crypt4gh-needed:\t crypt4gh\n",
            "    mem:\t mem\n",
            "/bin/bash: line 1: subread: command not found\n",
            "tree v2.0.2 (c) 1996 - 2022 by Steve Baker, Thomas Moore, Francesc Rocher, Florian Sesser, Kyosuke Tokoro \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title (ディレクトリ設定)\n",
        "# DIR_COLAB = \"/content\" #@param [\"/content\", \"option1\", \"option2\"] {allow-input: true}\n",
        "# DIR_RNAseq = \"/content/drive/MyDrive/RNAseq\" #@param [\"/content/drive/MyDrive/RNAseq\", \"option1\", \"option2\"] {allow-input: true}\n",
        "# DIR_RAWDATA = \"/content/drive/MyDrive/RNAseq/rawdata\" #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}"
      ],
      "metadata": {
        "id": "wBAYi6sBhobq"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title (ワーキングディレクトリ設定)\n",
        "# import os\n",
        "\n",
        "# # working directory path\n",
        "# DIR_WORK = \"/content/drive/MyDrive/RNAseq\" #@param [\"/content\", \"/content/drive/MyDrive/RNAseq\", \"/content/drive/MyDrive/RNAseq/rawdata\", \"option\"] {allow-input: true}\n",
        "\n",
        "# # %cd {DIR_RAWDATA}  # linux\n",
        "# os.chdir(DIR_WORK)\n",
        "# print(\"現在のディレクトリ:\", os.getcwd())"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1m-fH_CurLOk",
        "outputId": "521442a2-a662-4d6c-cffc-7acd82c92f9e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "現在のディレクトリ: /content/drive/MyDrive/RNAseq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title フォルダ作成: RNAseq以下に [bamfiles, downloads, featureCounts, multiqc, trimmed] を作成\n",
        "import os\n",
        "\n",
        "# working directory path\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq\" #@param [\"/content\", \"/content/drive/MyDrive/RNAseq\", \"/content/drive/MyDrive/RNAseq/rawdata\", \"option\"] {allow-input: true}\n",
        "\n",
        "# %cd {DIR_RAWDATA}  # linux\n",
        "os.chdir(DIR_WORK)\n",
        "print(\"現在のディレクトリ:\", os.getcwd())\n",
        "\n",
        "# folder name\n",
        "directories = [\"bamfiles\", \"downloads\", \"featureCounts\", \"multiqc\", \"trimmed\"]\n",
        "\n",
        "# create folder function\n",
        "def check_and_create_directory(directory):\n",
        "    if not os.path.exists(directory):\n",
        "        print(f\"Creating directory: {directory}\")\n",
        "        os.makedirs(directory)\n",
        "    else:\n",
        "        print(f\"Directory already exists: {directory}. Nothing to do.\")\n",
        "\n",
        "# create folder\n",
        "for directory in directories:\n",
        "    check_and_create_directory(directory)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f3Q6XfOhkH4c",
        "outputId": "9e4fa5bb-7b60-404e-9c8f-3dc9af00e6c8",
        "cellView": "form"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "現在のディレクトリ: /content/drive/MyDrive/RNAseq\n",
            "Directory already exists: bamfiles. Nothing to do.\n",
            "Directory already exists: downloads. Nothing to do.\n",
            "Directory already exists: featureCounts. Nothing to do.\n",
            "Directory already exists: multiqc. Nothing to do.\n",
            "Directory already exists: trimmed. Nothing to do.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title (フォルダとファイルの削除→必要があればアクティブにして実行)\n",
        "# import os\n",
        "# import shutil\n",
        "\n",
        "# 削除するディレクトリの親ディレクトリのパス\n",
        "# parent_directory = \"/content/drive/MyDrive/RNAseq\" #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# 削除するサブディレクトリのリスト\n",
        "# subdirectories = [\"downloads\", \"trimmed\", \"bamfiles\", \"featureCounts\", \"fastp\", \"fastp.1\", \"seqkit\"]\n",
        "\n",
        "# サブディレクトリとファイルを削除\n",
        "# for subdir in subdirectories:\n",
        "#     path = os.path.join(parent_directory, subdir)\n",
        "#     if os.path.isdir(path):\n",
        "#         os.rmdir(path)\n",
        "#         print(f\"{path} を削除しました。\")\n",
        "#     elif os.path.isfile(path):\n",
        "#         os.remove(path)\n",
        "#         print(f\"{path} を削除しました。\")\n",
        "#     else:\n",
        "#         print(f\"{path} は存在しません。\")\n",
        "\n",
        "# サブディレクトリを再帰的に削除\n",
        "# for subdir in subdirectories:\n",
        "#     path = os.path.join(parent_directory, subdir)\n",
        "#     if os.path.exists(path):\n",
        "#         if os.path.isdir(path):\n",
        "#             shutil.rmtree(path)\n",
        "#             print(f\"{path} を再帰的に削除しました。\")\n",
        "#         else:\n",
        "#             print(f\"{path} はディレクトリではないため削除できません。\")\n",
        "#     else:\n",
        "#         print(f\"{path} は存在しません。\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3w5AbDzEuWd3",
        "outputId": "ebfb7dd4-51c8-472f-85de-2d721b67290f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive/RNAseq/downloads は存在しません。\n",
            "/content/drive/MyDrive/RNAseq/trimmed は存在しません。\n",
            "/content/drive/MyDrive/RNAseq/bamfiles は存在しません。\n",
            "/content/drive/MyDrive/RNAseq/featureCounts は存在しません。\n",
            "/content/drive/MyDrive/RNAseq/fastp は存在しません。\n",
            "/content/drive/MyDrive/RNAseq/fastp.1 は存在しません。\n",
            "/content/drive/MyDrive/RNAseq/seqkit を再帰的に削除しました。\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title コンテンツ表示\n",
        "import os\n",
        "\n",
        "# ワーキングディレクトリ\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq\" #@param [\"/content\", \"/content/drive/MyDrive/RNAseq\", \"/content/drive/MyDrive/RNAseq/rawdata\", \"option\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリの移動\n",
        "os.chdir(DIR_WORK)\n",
        "print(\"現在のディレクトリ:\", os.getcwd())\n",
        "\n",
        "# def print_directory_tree(directory, indent=''):\n",
        "#     print(indent + os.path.basename(directory) + '/')\n",
        "#     indent += '    '\n",
        "#     for item in os.listdir(directory):\n",
        "#         item_path = os.path.join(directory, item)\n",
        "#         if os.path.isdir(item_path):\n",
        "#             print_directory_tree(item_path, indent)\n",
        "#         else:\n",
        "#             print(indent + item)\n",
        "\n",
        "# # 現在のワーキングディレクトリのツリー形式を表示\n",
        "# print_directory_tree(os.getcwd())\n",
        "\n",
        "\n",
        "\n",
        "## ----------シェルコマンドをGoogle Colaboratoryで実行----------\n",
        "# !apt-get -qq install tree # tree表示のためのツールをインストール\n",
        "!tree {DIR_WORK}\n",
        "# !tree {DIR_RAWDATA}\n",
        "## -------------------------------------------------------"
      ],
      "metadata": {
        "id": "GOcQq7mzizd0",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "outputId": "a8c75a78-3dd3-45ca-a12e-10c86e50b391"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "現在のディレクトリ: /content/drive/MyDrive/RNAseq\n",
            "\u001b[01;34m/content/drive/MyDrive/RNAseq\u001b[0m\n",
            "├── \u001b[01;34mbamfiles\u001b[0m\n",
            "│   ├── \u001b[00mHIP_PC10_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mHIP_PC10_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mHIP_PC30_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mHIP_PC30_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mHIP_PV_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mHIP_PV_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mHIP_SC10_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mHIP_SC10_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mHIP_SC30_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mHIP_SC30_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mHIP_SV_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mHIP_SV_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mNAc_PC10_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mNAc_PC10_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mNAc_PC30_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mNAc_PC30_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mNAc_PV_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mNAc_PV_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mNAc_SC10_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mNAc_SC10_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mNAc_SC30_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mNAc_SC30_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mNAc_SV_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mNAc_SV_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mPFC_PC10_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mPFC_PC10_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mPFC_PC30_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mPFC_PC30_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mPFC_PV_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mPFC_PV_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mPFC_SC10_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mPFC_SC10_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mPFC_SC30_sorted.bam\u001b[0m\n",
            "│   ├── \u001b[00mPFC_SC30_sorted.bam.bai\u001b[0m\n",
            "│   ├── \u001b[00mPFC_SV_sorted.bam\u001b[0m\n",
            "│   └── \u001b[00mPFC_SV_sorted.bam.bai\u001b[0m\n",
            "├── \u001b[01;34mdownloads\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic.fna\u001b[0m\n",
            "│   ├── \u001b[01;31mGCF_000001635.27_GRCm39_genomic.fna.gz\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic.gtf\u001b[0m\n",
            "│   ├── \u001b[01;31mGCF_000001635.27_GRCm39_genomic.gtf.gz\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.1.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.2.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.3.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.4.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.5.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.6.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.7.ht2\u001b[0m\n",
            "│   ├── \u001b[00mGCF_000001635.27_GRCm39_genomic_index.8.ht2\u001b[0m\n",
            "│   ├── \u001b[01;34mTrimmomatic-0.39\u001b[0m\n",
            "│   │   ├── \u001b[01;34madapters\u001b[0m\n",
            "│   │   │   ├── \u001b[00mNexteraPE-PE.fa\u001b[0m\n",
            "│   │   │   ├── \u001b[00mTruSeq2-PE.fa\u001b[0m\n",
            "│   │   │   ├── \u001b[00mTruSeq2-SE.fa\u001b[0m\n",
            "│   │   │   ├── \u001b[00mTruSeq3-PE-2.fa\u001b[0m\n",
            "│   │   │   ├── \u001b[00mTruSeq3-PE.fa\u001b[0m\n",
            "│   │   │   └── \u001b[00mTruSeq3-SE.fa\u001b[0m\n",
            "│   │   ├── \u001b[00mLICENSE\u001b[0m\n",
            "│   │   └── \u001b[00mtrimmomatic-0.39.jar\u001b[0m\n",
            "│   └── \u001b[01;31mTrimmomatic-0.39.zip\u001b[0m\n",
            "├── \u001b[01;34mfeatureCounts\u001b[0m\n",
            "│   ├── \u001b[00mfeatureCounts.txt\u001b[0m\n",
            "│   └── \u001b[00mfeatureCounts.txt.summary\u001b[0m\n",
            "├── \u001b[01;34mmultiqc\u001b[0m\n",
            "├── \u001b[01;34mR\u001b[0m\n",
            "│   ├── \u001b[00medgeR_result_bcv_0.1_test.txt\u001b[0m\n",
            "│   ├── \u001b[00mfeatureCounts_rm2to6.txt\u001b[0m\n",
            "│   ├── \u001b[00mfeatureCounts.txt\u001b[0m\n",
            "│   ├── \u001b[00mHIP_fdr.xlsx\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_HIP_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_HIP_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_HIP_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_HIP_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_HIP_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_NAc_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC10_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_HIP_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_HIP_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_HIP_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_HIP_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_NAc_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PC30_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_HIP_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_HIP_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_HIP_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_NAc_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_PV_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_HIP_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_HIP_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_NAc_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC10_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_HIP_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_NAc_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SC30_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_NAc_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_HIP_SV_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_NAc_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC10_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_NAc_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PC30_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_NAc_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_PV_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_NAc_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC10_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_NAc_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SC30_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SV_vs_PFC_PC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SV_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SV_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SV_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SV_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_NAc_SV_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC10_vs_PFC_PC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC10_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC10_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC10_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC10_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC30_vs_PFC_PV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC30_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC30_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PC30_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PV_vs_PFC_SC10.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PV_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_PV_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_SC10_vs_PFC_SC30.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_SC10_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mMA_plot_PFC_SC30_vs_PFC_SV.pdf\u001b[0m\n",
            "│   ├── \u001b[00mNAc_fdr.xlsx\u001b[0m\n",
            "│   └── \u001b[00mPFC_fdr.xlsx\u001b[0m\n",
            "├── \u001b[01;34mrawdata\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_PC10_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_PC10_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_PC30_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_PC30_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_PV_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_PV_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_SC10_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_SC10_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_SC30_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_SC30_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_SV_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mHIP_SV_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_PC10_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_PC10_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_PC30_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_PC30_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_PV_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_PV_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_SC10_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_SC10_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_SC30_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_SC30_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_SV_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mNAc_SV_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_PC10_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_PC10_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_PC30_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_PC30_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_PV_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_PV_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_SC10_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_SC10_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_SC30_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_SC30_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_SV_1.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[01;31mPFC_SV_2.fastq.gz\u001b[0m\n",
            "│   ├── \u001b[00msample_list_original.txt\u001b[0m\n",
            "│   └── \u001b[00msample_list.txt\u001b[0m\n",
            "└── \u001b[01;34mtrimmed\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_PC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_PC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_PC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_PC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_PV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_PV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_SC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_SC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_SC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_SC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_SV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_HIP_SV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_PC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_PC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_PC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_PC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_PV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_PV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_SC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_SC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_SC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_SC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_SV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_NAc_SV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_PC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_PC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_PC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_PC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_PV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_PV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_SC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_SC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_SC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_SC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_SV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31mpaired_PFC_SV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_PC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_PC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_PC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_PC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_PV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_PV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_SC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_SC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_SC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_SC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_SV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_HIP_SV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_PC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_PC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_PC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_PC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_PV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_PV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_SC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_SC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_SC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_SC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_SV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_NAc_SV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_PC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_PC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_PC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_PC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_PV_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_PV_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_SC10_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_SC10_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_SC30_1.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_SC30_2.trim.fastq.gz\u001b[0m\n",
            "    ├── \u001b[01;31munpaired_PFC_SV_1.trim.fastq.gz\u001b[0m\n",
            "    └── \u001b[01;31munpaired_PFC_SV_2.trim.fastq.gz\u001b[0m\n",
            "\n",
            "9 directories, 328 files\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "uVVVUw72_DZS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "GA1voWpmnKr6"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "X8KWWmXZHtwE"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# データ確認(SeqKit):未実装"
      ],
      "metadata": {
        "id": "nG9yEcd7CVHp"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title (seqkit: 実装できていない)\n",
        "# %cd /content\n",
        "\n",
        "# !git clone https://github.com/shenwei356/seqkit.git\n",
        "# %cd seqkit\n",
        "# !make\n",
        "# !sudo make install"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "moBuWaaqwCGf",
        "outputId": "23473be3-906f-4250-a443-617c9912fbf5",
        "cellView": "form"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n",
            "Cloning into 'seqkit'...\n",
            "remote: Enumerating objects: 10170, done.\u001b[K\n",
            "remote: Counting objects: 100% (2165/2165), done.\u001b[K\n",
            "remote: Compressing objects: 100% (683/683), done.\u001b[K\n",
            "remote: Total 10170 (delta 1406), reused 2140 (delta 1394), pack-reused 8005\u001b[K\n",
            "Receiving objects: 100% (10170/10170), 63.28 MiB | 22.40 MiB/s, done.\n",
            "Resolving deltas: 100% (6571/6571), done.\n",
            "/content/seqkit\n",
            "make: *** No targets specified and no makefile found.  Stop.\n",
            "make: *** No rule to make target 'install'.  Stop.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "## ----------シェルコマンドをPythonコードで実行----------\n",
        "# Trimmomaticをダウンロードして展開する\n",
        "# %cd /content/drive/MyDrive/RNAseq/downloads\n",
        "# !wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\n",
        "# !unzip Trimmomatic-0.39.zip\n",
        "## -------------------------------------------------------"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MkYqV8-6RSQZ",
        "outputId": "7b1f9b90-8977-4e31-8b01-b3bf89ad7c77"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive/RNAseq/downloads\n",
            "--2024-03-20 23:08:37--  http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\n",
            "Resolving www.usadellab.org (www.usadellab.org)... 80.83.125.111\n",
            "Connecting to www.usadellab.org (www.usadellab.org)|80.83.125.111|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 133596 (130K) [application/zip]\n",
            "Saving to: ‘Trimmomatic-0.39.zip’\n",
            "\n",
            "Trimmomatic-0.39.zi 100%[===================>] 130.46K   483KB/s    in 0.3s    \n",
            "\n",
            "2024-03-20 23:08:37 (483 KB/s) - ‘Trimmomatic-0.39.zip’ saved [133596/133596]\n",
            "\n",
            "Archive:  Trimmomatic-0.39.zip\n",
            "   creating: Trimmomatic-0.39/\n",
            "  inflating: Trimmomatic-0.39/LICENSE  \n",
            "  inflating: Trimmomatic-0.39/trimmomatic-0.39.jar  \n",
            "   creating: Trimmomatic-0.39/adapters/\n",
            "  inflating: Trimmomatic-0.39/adapters/NexteraPE-PE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq2-PE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq2-SE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq3-PE-2.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq3-PE.fa  \n",
            "  inflating: Trimmomatic-0.39/adapters/TruSeq3-SE.fa  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "NGdVs63TnzJ2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#トリミング(Trimmomatic)"
      ],
      "metadata": {
        "id": "WUuoz6xtCOuT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title ダウンロードトリミングツール: Trimmomatic\n",
        "import os\n",
        "import subprocess\n",
        "\n",
        "# ディレクトリ\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "DIR_DOWNLOADS = \"../downloads\"  #@param [\"../downloads\", \"/content/drive/MyDrive/RNAseq/downloads\", \"option1\"] {allow-input: true}\n",
        "DIR_TRIMMOMATIC = os.path.join(DIR_DOWNLOADS, \"Trimmomatic-0.39\")\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# すでにTrimmomaticが存在するかどうかをチェック\n",
        "if not os.path.exists(DIR_TRIMMOMATIC):\n",
        "    # TrimmomaticのZIPファイルをダウンロード\n",
        "    trimmomatic_zip_url = \"http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\"  #@param [\"http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\", \"option1\", \"option2\"] {allow-input: true}\n",
        "    trimmomatic_zip_file = os.path.basename(trimmomatic_zip_url)\n",
        "    subprocess.run([\"wget\", trimmomatic_zip_url], cwd=DIR_DOWNLOADS)\n",
        "\n",
        "    # TrimmomaticのZIPファイルを展開\n",
        "    subprocess.run([\"unzip\", trimmomatic_zip_file], cwd=DIR_DOWNLOADS)\n",
        "else:\n",
        "    print(\"Trimmomaticはすでに展開されています。\")\n",
        "\n",
        "\n",
        "## ----------シェルコマンドをPythonコードで実行----------\n",
        "# Trimmomaticをダウンロードして展開する\n",
        "# %cd /content/drive/MyDrive/RNAseq/downloads\n",
        "# !wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip\n",
        "# !unzip Trimmomatic-0.39.zip\n",
        "## -------------------------------------------------------\n"
      ],
      "metadata": {
        "cellView": "form",
        "id": "Oe7abMyFnzFe"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#@title トリミング（未実施サンプルデータのみ処理）: Trimmomatic\n",
        "import os\n",
        "import subprocess\n",
        "import time\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "sample_list_file = \"sample_list_original.txt\"  #@param [\"sample_list.txt\", \"sample_list_original.txt\", \"option1\"] {allow-input: true}\n",
        "java_file = \"../downloads/Trimmomatic-0.39/trimmomatic-0.39.jar\"  #@param [\"../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\", \"option1\", \"option2\"] {allow-input: true}\n",
        "illuminaclip_file = \"../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\"  #@param [\"../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# サンプルリストをファイルから読み取る\n",
        "with open(sample_list_file, \"r\") as f:\n",
        "    sample_list = f.read().splitlines()\n",
        "\n",
        "# サンプルリストをループで処理\n",
        "for sample_line in sample_list[1:]:  # 1行目をスキップして残りの行を処理\n",
        "    # サンプル名を分割して取得\n",
        "    sample_data = sample_line.strip().split(\"\\t\")\n",
        "    sample_number = sample_data[0]  # サンプル番号\n",
        "\n",
        "    # 出力ファイル名を検査してトリミング済みかどうかを確認\n",
        "    output_forward_paired = f\"../trimmed/paired_{sample_data[1]}_1.trim.fastq.gz\"\n",
        "    output_reverse_paired = f\"../trimmed/paired_{sample_data[1]}_2.trim.fastq.gz\"\n",
        "    if os.path.exists(output_forward_paired) and os.path.exists(output_reverse_paired):\n",
        "        print(f\"Sample No. {sample_number}, <{sample_data[1]}> has already been trimmed. Skipping...\")\n",
        "        continue\n",
        "\n",
        "    # 入力ファイルのフルパスを指定\n",
        "    input_forward = f\"{sample_data[1]}_1.fastq.gz\"\n",
        "    input_reverse = f\"{sample_data[1]}_2.fastq.gz\"\n",
        "    output_forward_paired = f\"../trimmed/paired_{sample_data[1]}_1.trim.fastq.gz\"\n",
        "    output_forward_unpaired = f\"../trimmed/unpaired_{sample_data[1]}_1.trim.fastq.gz\"\n",
        "    output_reverse_paired = f\"../trimmed/paired_{sample_data[1]}_2.trim.fastq.gz\"\n",
        "    output_reverse_unpaired = f\"../trimmed/unpaired_{sample_data[1]}_2.trim.fastq.gz\"\n",
        "\n",
        "    # ファイル名を出力して確認\n",
        "    print(f\"Sample No. {sample_number}, Forward Input File: {input_forward}, Reverse Input File: {input_reverse}\")\n",
        "    print(f\"Sample No. {sample_number}, Forward Output File: {output_forward_paired}, {output_forward_unpaired}, Reverse Output File: {output_reverse_paired}, {output_reverse_unpaired}\")\n",
        "\n",
        "    # 処理開始時間を記録\n",
        "    start_time = time.time()\n",
        "    print(f\"Processing started for sample {sample_data[1]} at {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))}\")\n",
        "\n",
        "    # コマンドを構築\n",
        "    trimmomatic_cmd = [\n",
        "        \"java\", \"-jar\", java_file, \"PE\", \"-phred33\",\n",
        "        input_forward, input_reverse,\n",
        "        output_forward_paired, output_forward_unpaired,\n",
        "        output_reverse_paired, output_reverse_unpaired,\n",
        "        f\"ILLUMINACLIP:{illuminaclip_file}\"\n",
        "    ]\n",
        "\n",
        "    # コマンド実行前のメッセージを表示\n",
        "    print(f\"Starting trimming for sample {sample_data[1]}...\")\n",
        "\n",
        "    # trimmomaticコマンドを実行\n",
        "    result = subprocess.run(trimmomatic_cmd, capture_output=True, text=True)\n",
        "\n",
        "    # 結果を表示\n",
        "    print(f\"Trimming for sample {sample_data[1]} finished with return code: {result.returncode}\")\n",
        "    if result.stdout:\n",
        "        print(result.stdout)\n",
        "    if result.stderr:\n",
        "        print(result.stderr)\n",
        "\n",
        "    # 処理終了時間を記録\n",
        "    end_time = time.time()\n",
        "\n",
        "    # 処理時間を計算して表示\n",
        "    processing_time = end_time - start_time\n",
        "    print(f\"Processing time for sample {sample_data[1]}: {processing_time} seconds\")\n",
        "\n",
        "    # 処理の区切りを示すメッセージを表示\n",
        "    print(\"------------------------------------------------------\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "nkthj2CgqAWM",
        "outputId": "b42cc313-45e1-4beb-fdd8-1e00d9065812"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Sample No. 1, <HIP_PC10> has already been trimmed. Skipping...\n",
            "Sample No. 2, <HIP_PC30> has already been trimmed. Skipping...\n",
            "Sample No. 3, <HIP_PV> has already been trimmed. Skipping...\n",
            "Sample No. 4, <HIP_SC10> has already been trimmed. Skipping...\n",
            "Sample No. 5, <HIP_SC30> has already been trimmed. Skipping...\n",
            "Sample No. 6, <HIP_SV> has already been trimmed. Skipping...\n",
            "Sample No. 7, <NAc_PC10> has already been trimmed. Skipping...\n",
            "Sample No. 8, <NAc_PC30> has already been trimmed. Skipping...\n",
            "Sample No. 9, <NAc_PV> has already been trimmed. Skipping...\n",
            "Sample No. 10, <NAc_SC10> has already been trimmed. Skipping...\n",
            "Sample No. 11, <NAc_SC30> has already been trimmed. Skipping...\n",
            "Sample No. 12, <NAc_SV> has already been trimmed. Skipping...\n",
            "Sample No. 13, <PFC_PC10> has already been trimmed. Skipping...\n",
            "Sample No. 14, <PFC_PC30> has already been trimmed. Skipping...\n",
            "Sample No. 15, <PFC_PV> has already been trimmed. Skipping...\n",
            "Sample No. 16, <PFC_SC10> has already been trimmed. Skipping...\n",
            "Sample No. 17, <PFC_SC30> has already been trimmed. Skipping...\n",
            "Sample No. 18, <PFC_SV> has already been trimmed. Skipping...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title (トリミング: 実行実績あり: Trimmomatic)\n",
        "\n",
        "import os\n",
        "import subprocess\n",
        "import time\n",
        "\n",
        "# ファイルパス\n",
        "%cd /content/drive/MyDrive/RNAseq/rawdata\n",
        "sample_list_file = \"sample_list.txt\"  #@param [\"sample_list.txt\", \"sample_list_original.txt\", \"option1\"] {allow-input: true}\n",
        "ILLUMINACLIP_file = \"../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\"  #@param [\"../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\", \"option1\", \"option2\"] {allow-input: true}\n",
        "JAVA_file = \"../downloads/Trimmomatic-0.39/trimmomatic-0.39.jar\"  #@param [\"../downloads/Trimmomatic-0.39/trimmomatic-0.39.jar\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# サンプルリストをファイルから読み取る\n",
        "with open(sample_list_file, \"r\") as f:\n",
        "    sample_list = f.read().splitlines()\n",
        "\n",
        "# サンプルリストをループで処理\n",
        "for sample_line in sample_list[1:]:  # 1行目をスキップして残りの行を処理\n",
        "    # サンプル名を分割して取得\n",
        "    sample_data = sample_line.strip().split(\"\\t\")\n",
        "    sample = sample_data[0]  # サンプル名\n",
        "    # 入力ファイルのフルパスを指定\n",
        "    input_forward = f\"{sample_data[1]}_1.fastq.gz\"\n",
        "    input_reverse = f\"{sample_data[1]}_2.fastq.gz\"\n",
        "    output_forward = f\"../trimmed/paired_{sample_data[1]}_1.trim.fastq.gz ../trimmed/unpaired_{sample_data[1]}_1.trim.fastq.gz \"\n",
        "    output_reverse = f\"../trimmed/paired_{sample_data[1]}_2.trim.fastq.gz ../trimmed/unpaired_{sample_data[1]}_2.trim.fastq.gz \"\n",
        "\n",
        "    # ファイル名を出力して確認\n",
        "    print(f\"Sample: {sample}, Forward Input File: {input_forward}, Reverse Input File: {input_reverse}\")\n",
        "    print(f\"Sample: {sample}, Forward Output File: {output_forward}, Reverse Output File: {output_reverse}\")\n",
        "\n",
        "    # 処理開始時間を記録\n",
        "    start_time = time.time()\n",
        "\n",
        "    # コマンドを構築\n",
        "    trimmomatic_cmd = f\"java -jar {JAVA_file} PE -phred33 {input_forward} {input_reverse} \" \\\n",
        "                          f\"{output_forward} {output_reverse} \" \\\n",
        "                          f\"ILLUMINACLIP:{ILLUMINACLIP_file}\"\n",
        "\n",
        "    # コマンド実行前のメッセージを表示\n",
        "    print(f\"Starting trimming for sample {sample_data[1]}...\")\n",
        "\n",
        "    # trimmomaticコマンドを実行\n",
        "    result = subprocess.run(trimmomatic_cmd, shell=True, capture_output=True, text=True)\n",
        "\n",
        "    # 結果を表示\n",
        "    print(f\"Trimming for sample {sample_data[1]} finished with return code: {result.returncode}\")\n",
        "    print(result.stdout)\n",
        "    print(result.stderr)\n",
        "\n",
        "    # 処理終了時間を記録\n",
        "    end_time = time.time()\n",
        "\n",
        "    # 処理時間を計算して表示\n",
        "    processing_time = end_time - start_time\n",
        "    print(f\"Processing time for sample {sample_data[1]}: {processing_time} seconds\")\n",
        "\n",
        "    # 処理の区切りを示すメッセージを表示\n",
        "    print(f\"------------------------------------------------------\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "J83kLdNjbz-0",
        "outputId": "d76501ca-a8b7-4b25-f7c6-9c5803a6c90f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive/RNAseq/rawdata\n",
            "Sample: 12, Forward Input File: NAc_SV_1.fastq.gz, Reverse Input File: NAc_SV_2.fastq.gz\n",
            "Sample: 12, Forward Output File: ../trimmed/paired_NAc_SV_1.trim.fastq.gz ../trimmed/unpaired_NAc_SV_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_NAc_SV_2.trim.fastq.gz ../trimmed/unpaired_NAc_SV_2.trim.fastq.gz \n",
            "Starting trimming for sample NAc_SV...\n",
            "Trimming for sample NAc_SV finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 NAc_SV_1.fastq.gz NAc_SV_2.fastq.gz ../trimmed/paired_NAc_SV_1.trim.fastq.gz ../trimmed/unpaired_NAc_SV_1.trim.fastq.gz ../trimmed/paired_NAc_SV_2.trim.fastq.gz ../trimmed/unpaired_NAc_SV_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 19831575 Both Surviving: 19779443 (99.74%) Forward Only Surviving: 52128 (0.26%) Reverse Only Surviving: 4 (0.00%) Dropped: 0 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample NAc_SV: 1324.9647433757782 seconds\n",
            "------------------------------------------------------\n",
            "Sample: 13, Forward Input File: PFC_PC10_1.fastq.gz, Reverse Input File: PFC_PC10_2.fastq.gz\n",
            "Sample: 13, Forward Output File: ../trimmed/paired_PFC_PC10_1.trim.fastq.gz ../trimmed/unpaired_PFC_PC10_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_PFC_PC10_2.trim.fastq.gz ../trimmed/unpaired_PFC_PC10_2.trim.fastq.gz \n",
            "Starting trimming for sample PFC_PC10...\n",
            "Trimming for sample PFC_PC10 finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 PFC_PC10_1.fastq.gz PFC_PC10_2.fastq.gz ../trimmed/paired_PFC_PC10_1.trim.fastq.gz ../trimmed/unpaired_PFC_PC10_1.trim.fastq.gz ../trimmed/paired_PFC_PC10_2.trim.fastq.gz ../trimmed/unpaired_PFC_PC10_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 26098319 Both Surviving: 26041565 (99.78%) Forward Only Surviving: 56749 (0.22%) Reverse Only Surviving: 4 (0.00%) Dropped: 1 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample PFC_PC10: 1705.8579213619232 seconds\n",
            "------------------------------------------------------\n",
            "Sample: 14, Forward Input File: PFC_PC30_1.fastq.gz, Reverse Input File: PFC_PC30_2.fastq.gz\n",
            "Sample: 14, Forward Output File: ../trimmed/paired_PFC_PC30_1.trim.fastq.gz ../trimmed/unpaired_PFC_PC30_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_PFC_PC30_2.trim.fastq.gz ../trimmed/unpaired_PFC_PC30_2.trim.fastq.gz \n",
            "Starting trimming for sample PFC_PC30...\n",
            "Trimming for sample PFC_PC30 finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 PFC_PC30_1.fastq.gz PFC_PC30_2.fastq.gz ../trimmed/paired_PFC_PC30_1.trim.fastq.gz ../trimmed/unpaired_PFC_PC30_1.trim.fastq.gz ../trimmed/paired_PFC_PC30_2.trim.fastq.gz ../trimmed/unpaired_PFC_PC30_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 20440573 Both Surviving: 20359679 (99.60%) Forward Only Surviving: 80888 (0.40%) Reverse Only Surviving: 3 (0.00%) Dropped: 3 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample PFC_PC30: 1346.092553138733 seconds\n",
            "------------------------------------------------------\n",
            "Sample: 15, Forward Input File: PFC_PV_1.fastq.gz, Reverse Input File: PFC_PV_2.fastq.gz\n",
            "Sample: 15, Forward Output File: ../trimmed/paired_PFC_PV_1.trim.fastq.gz ../trimmed/unpaired_PFC_PV_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_PFC_PV_2.trim.fastq.gz ../trimmed/unpaired_PFC_PV_2.trim.fastq.gz \n",
            "Starting trimming for sample PFC_PV...\n",
            "Trimming for sample PFC_PV finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 PFC_PV_1.fastq.gz PFC_PV_2.fastq.gz ../trimmed/paired_PFC_PV_1.trim.fastq.gz ../trimmed/unpaired_PFC_PV_1.trim.fastq.gz ../trimmed/paired_PFC_PV_2.trim.fastq.gz ../trimmed/unpaired_PFC_PV_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 27363579 Both Surviving: 27308768 (99.80%) Forward Only Surviving: 54807 (0.20%) Reverse Only Surviving: 3 (0.00%) Dropped: 1 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample PFC_PV: 1779.9007580280304 seconds\n",
            "------------------------------------------------------\n",
            "Sample: 16, Forward Input File: PFC_SC10_1.fastq.gz, Reverse Input File: PFC_SC10_2.fastq.gz\n",
            "Sample: 16, Forward Output File: ../trimmed/paired_PFC_SC10_1.trim.fastq.gz ../trimmed/unpaired_PFC_SC10_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_PFC_SC10_2.trim.fastq.gz ../trimmed/unpaired_PFC_SC10_2.trim.fastq.gz \n",
            "Starting trimming for sample PFC_SC10...\n",
            "Trimming for sample PFC_SC10 finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 PFC_SC10_1.fastq.gz PFC_SC10_2.fastq.gz ../trimmed/paired_PFC_SC10_1.trim.fastq.gz ../trimmed/unpaired_PFC_SC10_1.trim.fastq.gz ../trimmed/paired_PFC_SC10_2.trim.fastq.gz ../trimmed/unpaired_PFC_SC10_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 27663790 Both Surviving: 27623088 (99.85%) Forward Only Surviving: 40700 (0.15%) Reverse Only Surviving: 2 (0.00%) Dropped: 0 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample PFC_SC10: 1987.8906688690186 seconds\n",
            "------------------------------------------------------\n",
            "Sample: 17, Forward Input File: PFC_SC30_1.fastq.gz, Reverse Input File: PFC_SC30_2.fastq.gz\n",
            "Sample: 17, Forward Output File: ../trimmed/paired_PFC_SC30_1.trim.fastq.gz ../trimmed/unpaired_PFC_SC30_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_PFC_SC30_2.trim.fastq.gz ../trimmed/unpaired_PFC_SC30_2.trim.fastq.gz \n",
            "Starting trimming for sample PFC_SC30...\n",
            "Trimming for sample PFC_SC30 finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 PFC_SC30_1.fastq.gz PFC_SC30_2.fastq.gz ../trimmed/paired_PFC_SC30_1.trim.fastq.gz ../trimmed/unpaired_PFC_SC30_1.trim.fastq.gz ../trimmed/paired_PFC_SC30_2.trim.fastq.gz ../trimmed/unpaired_PFC_SC30_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 25770369 Both Surviving: 25714296 (99.78%) Forward Only Surviving: 56069 (0.22%) Reverse Only Surviving: 2 (0.00%) Dropped: 2 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample PFC_SC30: 1887.9097743034363 seconds\n",
            "------------------------------------------------------\n",
            "Sample: 18, Forward Input File: PFC_SV_1.fastq.gz, Reverse Input File: PFC_SV_2.fastq.gz\n",
            "Sample: 18, Forward Output File: ../trimmed/paired_PFC_SV_1.trim.fastq.gz ../trimmed/unpaired_PFC_SV_1.trim.fastq.gz , Reverse Output File: ../trimmed/paired_PFC_SV_2.trim.fastq.gz ../trimmed/unpaired_PFC_SV_2.trim.fastq.gz \n",
            "Starting trimming for sample PFC_SV...\n",
            "Trimming for sample PFC_SV finished with return code: 0\n",
            "\n",
            "TrimmomaticPE: Started with arguments:\n",
            " -phred33 PFC_SV_1.fastq.gz PFC_SV_2.fastq.gz ../trimmed/paired_PFC_SV_1.trim.fastq.gz ../trimmed/unpaired_PFC_SV_1.trim.fastq.gz ../trimmed/paired_PFC_SV_2.trim.fastq.gz ../trimmed/unpaired_PFC_SV_2.trim.fastq.gz ILLUMINACLIP:../downloads/Trimmomatic-0.39/adapters/TruSeq2-PE.fa:2:30:10\n",
            "Multiple cores found: Using 2 threads\n",
            "Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT'\n",
            "Using Long Clipping Sequence: 'AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTAATGATACGGCGACCACCGAGATCTACAC'\n",
            "Using Long Clipping Sequence: 'TTTTTTTTTTCAAGCAGAAGACGGCATACGA'\n",
            "Using Long Clipping Sequence: 'CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT'\n",
            "Using Long Clipping Sequence: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'\n",
            "ILLUMINACLIP: Using 1 prefix pairs, 6 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
            "Input Read Pairs: 21901183 Both Surviving: 21849035 (99.76%) Forward Only Surviving: 52145 (0.24%) Reverse Only Surviving: 3 (0.00%) Dropped: 0 (0.00%)\n",
            "TrimmomaticPE: Completed successfully\n",
            "\n",
            "Processing time for sample PFC_SV: 1445.2629964351654 seconds\n",
            "------------------------------------------------------\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "ksieIJ2HqAfP"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#マッピング(HISAT2)"
      ],
      "metadata": {
        "id": "I-MXP3MgCI8r"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title ゲノム配列情報（FNA）のダウンロード: HISAT2\n",
        "import gzip\n",
        "import os\n",
        "import shutil\n",
        "import subprocess\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "DIR_DOWNLOADS = \"../downloads\"  #@param [\"../downloads\", \"/content/drive/MyDrive/RNAseq/downloads\", \"option1\"] {allow-input: true}\n",
        "fna_file = \"GCF_000001635.27_GRCm39_genomic.fna\"  #@param [\"GCF_000001635.27_GRCm39_genomic.fna.gz\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# ファイルが存在するかチェック\n",
        "if not os.path.exists(os.path.join(DIR_DOWNLOADS, fna_file)):\n",
        "    # ダウンロード\n",
        "    fna_file_url = \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_genomic.fna.gz\"  #@param [\"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_genomic.fna.gz\", \"option1\", \"option2\"] {allow-input: true}\n",
        "    subprocess.run([\"wget\", fna_file_url, \"-P\", DIR_DOWNLOADS])\n",
        "\n",
        "    # 解凍\n",
        "    fna_file_path = os.path.join(DIR_DOWNLOADS, fna_file + \".gz\")\n",
        "    output_file_path = os.path.join(DIR_DOWNLOADS, fna_file)\n",
        "    with gzip.open(fna_file_path, 'rb') as f_in:\n",
        "        with open(output_file_path, 'wb') as f_out:\n",
        "            shutil.copyfileobj(f_in, f_out)\n",
        "else:\n",
        "    print(\"ファイルがすでに存在します。\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "g7YU1uV-l4vs",
        "outputId": "247a1c65-ba89-4a89-a993-e37cea9a45e8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "ファイルがすでに存在します。\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title （インストール: HISAT2）\n",
        "# import os\n",
        "# !apt-get -qq install hisat2\n",
        "# !apt-get -qq install samtools\n",
        "# os.environ['PATH'] += \":/usr/local/bin\"\n",
        "# !hisat2 --version\n",
        "# !samtools --version"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "yol4zwJO4qJb",
        "outputId": "360cbf43-21a6-4c51-a38b-a72048e0f5e8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "hisat2 is already the newest version (2.2.1-3).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 39 not upgraded.\n",
            "/usr/bin/hisat2-align-s version 2.2.1\n",
            "64-bit\n",
            "Built on Debian\n",
            "28 September 2021\n",
            "Compiler: gcc version 11.2.0 (Ubuntu 11.2.0-7ubuntu2) \n",
            "Options: -O3   -funroll-loops -g3 -Wdate-time -D_FORTIFY_SOURCE=2 -std=c++11\n",
            "Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title インデックスの作成: HISAT2\n",
        "import os\n",
        "import subprocess\n",
        "import time\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "genomic_fna = \"../downloads/GCF_000001635.27_GRCm39_genomic.fna\"  #@param [\"../downloads/GCF_000001635.27_GRCm39_genomic.fna\", \"option1\", \"option2\"] {allow-input: true}\n",
        "index_path = \"../downloads/GCF_000001635.27_GRCm39_genomic_index\"  #@param [\"../downloads/GCF_000001635.27_GRCm39_genomic_index\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# 処理開始時間を記録\n",
        "start_time = time.time()\n",
        "print(f\"Processing started at {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))}\")\n",
        "\n",
        "# インデックスファイルが存在するかチェック\n",
        "if not os.path.exists(index_path + \".1.ht2\"):\n",
        "    # インデックスを作成\n",
        "    print(\"Creating index...\")\n",
        "    subprocess.run([\"hisat2-build\", genomic_fna, index_path])\n",
        "    print(\"Index creation completed.\")\n",
        "else:\n",
        "    print(\"インデックスファイルはすでに存在します。\")\n",
        "\n",
        "# 処理終了時間を記録\n",
        "end_time = time.time()\n",
        "\n",
        "# 処理時間を計算して表示\n",
        "processing_time = end_time - start_time\n",
        "print(f\"Processing time: {processing_time} seconds.\")\n",
        "\n",
        "# 2h 5m 2s\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iQEmZfKBnTYu",
        "outputId": "d9634ec9-a7f9-4a4f-97ba-1dc250b1b65c",
        "cellView": "form"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Creating index...\n",
            "Index creation completed.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title マッピング（未実施サンプルデータのみ処理）: HISAT2\n",
        "import locale\n",
        "import os\n",
        "import subprocess\n",
        "import time\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "sample_list_file = \"sample_list_original.txt\"  #@param [\"sample_list.txt\", \"sample_list_original.txt\", \"option1\"] {allow-input: true}\n",
        "index_path = \"../downloads/GCF_000001635.27_GRCm39_genomic_index\"  #@param [\"../downloads/GCF_000001635.27_GRCm39_genomic_index\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# サンプルリストファイルからサンプルを取得し、HISAT2を実行\n",
        "with open(sample_list_file, \"r\") as f:\n",
        "    for line in f.readlines()[1:]:\n",
        "        sample = line.strip().split()[1]\n",
        "        print(f\"Processing sample: {sample}...\")\n",
        "        file1 = f\"../trimmed/paired_{sample}_1.trim.fastq.gz\"\n",
        "        file2 = f\"../trimmed/paired_{sample}_2.trim.fastq.gz\"\n",
        "        bam_file = f\"../bamfiles/{sample}_sorted.bam\"\n",
        "\n",
        "        # 処理開始時間を記録\n",
        "        start_time = time.time()\n",
        "        print(f\"Processing started for sample {sample} at {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))}\")\n",
        "\n",
        "        # すでに処理されたサンプルかどうかを確認\n",
        "        if os.path.exists(bam_file):\n",
        "            print(f\"Sample {sample} has already been processed. Skipping...\")\n",
        "            continue\n",
        "\n",
        "        # HISAT2を実行\n",
        "        hisat2_cmd = [\n",
        "            \"hisat2\", \"-p\", \"1\", \"-x\", index_path, \"-1\", file1, \"-2\", file2\n",
        "        ]\n",
        "        hisat2_proc = subprocess.Popen(hisat2_cmd, stdout=subprocess.PIPE)\n",
        "\n",
        "        # samtoolsを使ってバムファイルに変換\n",
        "        samtools_cmd = [\n",
        "            \"samtools\", \"sort\", \"-@ 1\", \"-\", \"-o\", bam_file\n",
        "        ]\n",
        "        samtools_proc = subprocess.Popen(samtools_cmd, stdin=hisat2_proc.stdout)\n",
        "        samtools_proc.communicate()\n",
        "\n",
        "        # バムファイルにインデックスを作成\n",
        "        subprocess.run([\"samtools\", \"index\", \"-@\", \"1\", bam_file])\n",
        "\n",
        "        # 処理終了時間を記録\n",
        "        end_time = time.time()\n",
        "\n",
        "        # 処理時間を計算して表示\n",
        "        processing_time = end_time - start_time\n",
        "        print(f\"Sample {sample} processing complete. Processing time: {processing_time} seconds.\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zz7N4xpkurvj",
        "outputId": "c1690446-5430-4839-870c-ee7b49691415",
        "cellView": "form"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Processing sample: HIP_PC10...\n",
            "Processing started for sample HIP_PC10 at 2024-03-21 23:19:56\n",
            "Sample HIP_PC10 has already been processed. Skipping...\n",
            "Processing sample: HIP_PC30...\n",
            "Processing started for sample HIP_PC30 at 2024-03-21 23:19:56\n",
            "Sample HIP_PC30 has already been processed. Skipping...\n",
            "Processing sample: HIP_PV...\n",
            "Processing started for sample HIP_PV at 2024-03-21 23:19:56\n",
            "Sample HIP_PV has already been processed. Skipping...\n",
            "Processing sample: HIP_SC10...\n",
            "Processing started for sample HIP_SC10 at 2024-03-21 23:19:56\n",
            "Sample HIP_SC10 has already been processed. Skipping...\n",
            "Processing sample: HIP_SC30...\n",
            "Processing started for sample HIP_SC30 at 2024-03-21 23:19:56\n",
            "Sample HIP_SC30 has already been processed. Skipping...\n",
            "Processing sample: HIP_SV...\n",
            "Processing started for sample HIP_SV at 2024-03-21 23:19:56\n",
            "Sample HIP_SV has already been processed. Skipping...\n",
            "Processing sample: NAc_PC10...\n",
            "Processing started for sample NAc_PC10 at 2024-03-21 23:19:56\n",
            "Sample NAc_PC10 processing complete. Processing time: 3040.9675855636597 seconds.\n",
            "Processing sample: NAc_PC30...\n",
            "Processing started for sample NAc_PC30 at 2024-03-22 00:10:37\n",
            "Sample NAc_PC30 has already been processed. Skipping...\n",
            "Processing sample: NAc_PV...\n",
            "Processing started for sample NAc_PV at 2024-03-22 00:10:37\n",
            "Sample NAc_PV has already been processed. Skipping...\n",
            "Processing sample: NAc_SC10...\n",
            "Processing started for sample NAc_SC10 at 2024-03-22 00:10:37\n",
            "Sample NAc_SC10 has already been processed. Skipping...\n",
            "Processing sample: NAc_SC30...\n",
            "Processing started for sample NAc_SC30 at 2024-03-22 00:10:37\n",
            "Sample NAc_SC30 has already been processed. Skipping...\n",
            "Processing sample: NAc_SV...\n",
            "Processing started for sample NAc_SV at 2024-03-22 00:10:37\n",
            "Sample NAc_SV has already been processed. Skipping...\n",
            "Processing sample: PFC_PC10...\n",
            "Processing started for sample PFC_PC10 at 2024-03-22 00:10:37\n",
            "Sample PFC_PC10 has already been processed. Skipping...\n",
            "Processing sample: PFC_PC30...\n",
            "Processing started for sample PFC_PC30 at 2024-03-22 00:10:37\n",
            "Sample PFC_PC30 processing complete. Processing time: 3100.0796914100647 seconds.\n",
            "Processing sample: PFC_PV...\n",
            "Processing started for sample PFC_PV at 2024-03-22 01:02:17\n",
            "Sample PFC_PV processing complete. Processing time: 4078.9758954048157 seconds.\n",
            "Processing sample: PFC_SC10...\n",
            "Processing started for sample PFC_SC10 at 2024-03-22 02:10:16\n",
            "Sample PFC_SC10 processing complete. Processing time: 4150.094062805176 seconds.\n",
            "Processing sample: PFC_SC30...\n",
            "Processing started for sample PFC_SC30 at 2024-03-22 03:19:26\n",
            "Sample PFC_SC30 processing complete. Processing time: 3777.558183670044 seconds.\n",
            "Processing sample: PFC_SV...\n",
            "Processing started for sample PFC_SV at 2024-03-22 04:22:24\n",
            "Sample PFC_SV processing complete. Processing time: 3185.1986832618713 seconds.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "t_XQnpskdeqa"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#カウント(featureCount(subread))"
      ],
      "metadata": {
        "id": "Mtuypeo9B009"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#@title アノテーション（GTF）のダウンロード: featureCount(subread)\n",
        "import gzip\n",
        "import os\n",
        "import shutil\n",
        "import subprocess\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "DIR_DOWNLOADS = \"../downloads\"  #@param [\"../download\", \"/content/drive/MyDrive/RNAseq/downloads\", \"option1\"] {allow-input: true}\n",
        "gtf_file = \"GCF_000001635.27_GRCm39_genomic.gtf\"  #@param [\"GCF_000001635.27_GRCm39_genomic.gtf\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# ファイルが存在するかチェック\n",
        "if not os.path.exists(os.path.join(DIR_DOWNLOADS, gtf_file)):\n",
        "    # ダウンロード\n",
        "    gtf_file_url = \"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_genomic.gtf.gz\"  #@param [\"https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39/GCF_000001635.27_GRCm39_genomic.gtf.gz\", \"option1\", \"option2\"] {allow-input: true}\n",
        "    subprocess.run([\"wget\", gtf_file_url, \"-P\", DIR_DOWNLOADS])\n",
        "\n",
        "    # 解凍\n",
        "    gtf_file_path = os.path.join(DIR_DOWNLOADS, gtf_file + \".gz\")\n",
        "    output_file_path = os.path.join(DIR_DOWNLOADS, gtf_file)\n",
        "    with gzip.open(gtf_file_path, 'rb') as f_in:\n",
        "        with open(output_file_path, 'wb') as f_out:\n",
        "            shutil.copyfileobj(f_in, f_out)\n",
        "else:\n",
        "    print(\"ファイルがすでに存在します。\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "sl3tZuiy636u",
        "outputId": "5be09631-b891-42d1-8486-1b2a7787e360"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "ファイルがすでに存在します。\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title リードカウント: featureCount(subread)\n",
        "import os\n",
        "import subprocess\n",
        "import time\n",
        "!apt-get -qq install subread\n",
        "os.environ['PATH'] += \":/usr/local/bin\"\n",
        "\n",
        "# ワーキングディレクトリ\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# ファイルパス\n",
        "annotation_file = \"/content/drive/MyDrive/RNAseq/downloads/GCF_000001635.27_GRCm39_genomic.gtf\"  #@param [\"/content/drive/MyDrive/RNAseq/downloads/GCF_000001635.27_GRCm39_genomic.gtf\", \"../downloads/GCF_000001635.27_GRCm39_genomic.gtf\", \"option1\", \"option2\"] {allow-input: true}\n",
        "input_file = \"/content/drive/MyDrive/RNAseq/bamfiles/*_sorted.bam\" #@param [\"/content/drive/MyDrive/RNAseq/bamfiles/*_sorted.bam\", \"../bamfiles/*_sorted.bam\", \"option1\", \"option2\"] {allow-input: true}\n",
        "output_file = \"/content/drive/MyDrive/RNAseq/featureCounts/featureCounts.txt\"  #@param [\"/content/drive/MyDrive/RNAseq/featureCounts/featureCounts.txt\", \"../featureCounts/featureCounts.txt\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# featureCountsの定義\n",
        "# featureCounts_cmd = [\n",
        "#     \"featureCounts\",\n",
        "#     \"-F\", \"GTF\",\n",
        "#     \"-t\", \"CDS\",\n",
        "#     \"-g\", \"gene_id\",\n",
        "#     \"-a\", annotation_file,\n",
        "#     \"-o\", output_file,\n",
        "#     \"-p\", input_file\n",
        "# ]\n",
        "\n",
        "# 実行時間を記録\n",
        "start_time = time.time()\n",
        "print(f\"Processing started at {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))}\")\n",
        "\n",
        "# featureCountsを実行\n",
        "# subprocess.run(featureCounts_cmd)\n",
        "!featureCounts -F GTF -t CDS -g gene_id \\\n",
        "    -a $annotation_file \\\n",
        "    -o $output_file \\\n",
        "    -p $input_file\n",
        "\n",
        "# 実行時間を計算して表示\n",
        "end_time = time.time()\n",
        "execution_time = end_time - start_time\n",
        "print(f\"Processing time: {execution_time} seconds.\")\n"
      ],
      "metadata": {
        "id": "ITFO1M7UdecX",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "outputId": "08279c3f-7a01-4b1d-edda-5e7a6f746cc2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Processing started at 2024-03-22 06:00:37\n",
            "\n",
            "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
            "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
            "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
            "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
            "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
            "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
            "\t  v2.0.3\n",
            "\n",
            "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
            "||  \u001b[0m                                                                          ||\n",
            "||             Input files : \u001b[36m18 BAM files  \u001b[0m \u001b[0m                                  ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "||                           \u001b[36mHIP_PC10_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mHIP_PC30_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mHIP_PV_sorted.bam\u001b[0m \u001b[0m                               ||\n",
            "||                           \u001b[36mHIP_SC10_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mHIP_SC30_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mHIP_SV_sorted.bam\u001b[0m \u001b[0m                               ||\n",
            "||                           \u001b[36mNAc_PC10_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mNAc_PC30_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mNAc_PV_sorted.bam\u001b[0m \u001b[0m                               ||\n",
            "||                           \u001b[36mNAc_SC10_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mNAc_SC30_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mNAc_SV_sorted.bam\u001b[0m \u001b[0m                               ||\n",
            "||                           \u001b[36mPFC_PC10_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mPFC_PC30_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mPFC_PV_sorted.bam\u001b[0m \u001b[0m                               ||\n",
            "||                           \u001b[36mPFC_SC10_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mPFC_SC30_sorted.bam\u001b[0m \u001b[0m                             ||\n",
            "||                           \u001b[36mPFC_SV_sorted.bam\u001b[0m \u001b[0m                               ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "||             Output file : \u001b[36mfeatureCounts.txt\u001b[0m \u001b[0m                               ||\n",
            "||                 Summary : \u001b[36mfeatureCounts.txt.summary\u001b[0m \u001b[0m                       ||\n",
            "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
            "||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
            "||              Annotation : \u001b[36mGCF_000001635.27_GRCm39_genomic.gtf (GTF)\u001b[0m \u001b[0m       ||\n",
            "||      Dir for temp files : \u001b[36m/content/drive/MyDrive/RNAseq/featureCounts\u001b[0m \u001b[0m     ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
            "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
            "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
            "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
            "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "\\\\============================================================================//\n",
            "\n",
            "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Load annotation file GCF_000001635.27_GRCm39_genomic.gtf \u001b[0m... \u001b[0m              ||\n",
            "||    Features : \u001b[36m1203872\u001b[0m \u001b[0m                                                     ||\n",
            "||    Meta-features : \u001b[36m22864\u001b[0m \u001b[0m                                                  ||\n",
            "||    Chromosomes/contigs : \u001b[36m37\u001b[0m \u001b[0m                                               ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file HIP_PC10_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m55577538\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m25720570 (46.3%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.39 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file HIP_PC30_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m60001478\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m27750607 (46.2%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.73 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file HIP_PV_sorted.bam... \u001b[0m                                     ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m43865592\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m20188707 (46.0%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.33 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file HIP_SC10_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m45366814\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m20949272 (46.2%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.36 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file HIP_SC30_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m46528095\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m21439118 (46.1%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.31 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file HIP_SV_sorted.bam... \u001b[0m                                     ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m48761490\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m22402615 (45.9%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.36 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file NAc_PC10_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m44738712\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m19950897 (44.6%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.30 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file NAc_PC30_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m43419295\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m18677391 (43.0%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.19 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file NAc_PV_sorted.bam... \u001b[0m                                     ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m47450168\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m20577497 (43.4%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.28 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file NAc_SC10_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m60321421\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m25708482 (42.6%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.64 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file NAc_SC30_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m50051481\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m21603334 (43.2%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.57 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file NAc_SV_sorted.bam... \u001b[0m                                     ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m43399791\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m19092737 (44.0%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.19 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file PFC_PC10_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m56650508\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m25111550 (44.3%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m2.40 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file PFC_PC30_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m44713217\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m19733190 (44.1%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.26 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file PFC_PV_sorted.bam... \u001b[0m                                     ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m59509216\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m26160120 (44.0%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.60 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file PFC_SC10_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m59856529\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m26497209 (44.3%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.61 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file PFC_SC30_sorted.bam... \u001b[0m                                   ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m55934348\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m25174651 (45.0%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.69 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Process BAM file PFC_SV_sorted.bam... \u001b[0m                                     ||\n",
            "||    Paired-end reads are included. \u001b[0m                                         ||\n",
            "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
            "||    Total alignments : \u001b[36m47321181\u001b[0m \u001b[0m                                            ||\n",
            "||    Successfully assigned alignments : \u001b[36m21332952 (45.1%)\u001b[0m \u001b[0m                    ||\n",
            "||    Running time : \u001b[36m1.30 minutes\u001b[0m \u001b[0m                                            ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Write the final count table. \u001b[0m                                              ||\n",
            "|| Write the read assignment summary. \u001b[0m                                        ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "|| Summary of counting results can be found in file \"/content/drive/MyDrive/ \u001b[0m ||\n",
            "|| RNAseq/featureCounts/featureCounts.txt.summary\" \u001b[0m                           ||\n",
            "||  \u001b[0m                                                                          ||\n",
            "\\\\============================================================================//\n",
            "\n",
            "Processing time: 1601.8713212013245 seconds.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "_qZclYQMnLGl"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#(クオリティコントロールレポート(MultiQC): リポートが出力されない)"
      ],
      "metadata": {
        "id": "LNpoEMHkDy3H"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content\"  #@param [\"/content\", \"/content/drive/MyDrive/RNAseq/rawdata\", \"option2\"] {allow-input: true}\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "!git clone https://github.com/MultiQC/MultiQC_Notebook.git\n",
        "\n"
      ],
      "metadata": {
        "id": "Nh4W-j4z8SqZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# %pip install --force-reinstall --upgrade git+https://github.com/ewels/MultiQC.git"
      ],
      "metadata": {
        "id": "PwplxSYT_FOk"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import multiqc\n",
        "print(multiqc.__version__)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AXTm0lR6_GSv",
        "outputId": "b7e707b4-81a6-47cf-d9d7-7d3d59894f45"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1.22.dev0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# (クオリティコントロールレポート: MultiQC): リポートが出力されない\n",
        "# %pip install --force-reinstall --upgrade git+https://github.com/ewels/MultiQC.git\n",
        "\n",
        "import os\n",
        "import subprocess\n",
        "import time\n",
        "import multiqc\n",
        "print(multiqc.__version__)\n",
        "\n",
        "# ワーキングディレクトリとファイルパス\n",
        "DIR_WORK = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\", \"option2\"] {allow-input: true}\n",
        "DIR_MULTIQC = \"/content/drive/MyDrive/RNAseq/multiqc\"  #@param [\"../multiqc\", \"/content/drive/MyDrive/RNAseq/multiqc\", \"option1\"] {allow-input: true}\n",
        "\n",
        "# 入力ディレクトリのリスト\n",
        "input_path_1 = \"/content/drive/MyDrive/RNAseq/rawdata\"  #@param [\".\", \"/content/drive/MyDrive/RNAseq/rawdata\", \"option1\"] {allow-input: true}\n",
        "input_path_2 = \"/content/drive/MyDrive/RNAseq/trimmed\"  #@param [\"../trimmed\", \"/content/drive/MyDrive/RNAseq/trimmed\", \"option1\"] {allow-input: true}\n",
        "input_path_3 = \"\"  #@param [\"\", \"option1\", \"option2\"] {allow-input: true}\n",
        "\n",
        "DIR_INPUT = [input_path_1, input_path_2, input_path_3]\n",
        "DIR_ANALYSIS = DIR_INPUT\n",
        "\n",
        "# ワーキングディレクトリへ移動\n",
        "os.chdir(DIR_WORK)\n",
        "\n",
        "# 処理開始時間を記録\n",
        "start_time = time.time()\n",
        "print(f\"Processing started at {time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))}\")\n",
        "\n",
        "# MultiQCを実行\n",
        "multiqc.run(\n",
        "            # input_dir= DIR_INPUT,\n",
        "            output_dir=DIR_MULTIQC,\n",
        "            analysis_dir=DIR_ANALYSIS)\n",
        "\n",
        "print(\"MultiQC processing completed.\")\n",
        "\n",
        "# 処理終了時間を記録\n",
        "end_time = time.time()\n",
        "\n",
        "# 処理時間を計算して表示\n",
        "processing_time = end_time - start_time\n",
        "print(f\"Processing time: {processing_time} seconds.\")\n"
      ],
      "metadata": {
        "id": "zpZ9MdGX_GKS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import IPython\n",
        "\n",
        "# Best if using Google Colab\n",
        "IPython.display.HTML(filename='./multiqc_report.html')\n",
        "\n",
        "# Best if running locally\n",
        "# IPython.display.IFrame('./multiqc_report.html', '100%', 600)"
      ],
      "metadata": {
        "id": "n4EugWsx_GGE"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "NJfAdaoE_DcB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "knA2q1sdKEjZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#その他"
      ],
      "metadata": {
        "id": "eBgQSaJ7CBTE"
      }
    }
  ]
}